learning/katas/go/Core Transforms/Side Input/Side Input/pkg/task/task.go

```diff
@@ -0,0 +1,45 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package task
+
+import (
+	"github.com/apache/beam/sdks/go/pkg/beam"
+)
+
+type Person struct {
+	Name, City, Country string
+}
+
+func ApplyTransform(s beam.Scope, personsKV beam.PCollection, citiesToCountries beam.PCollection) beam.PCollection {
+	citiesToCountriesView := beam.SideInput{
+		Input: citiesToCountries,
+	}
+	return beam.ParDo(s, joinFn, personsKV, citiesToCountriesView)
+}
+
+func joinFn(person Person, citiesToCountriesIter func(*string, *string) bool, emit func(Person)) {
+	var city, country string
+	for citiesToCountriesIter(&city, &country) {
```

Comment:
Is there a way to use a KV-like API (e.g. get(key)) instead of iterating each of the KV pair?
The way this works seems like a List collection with a tuple.
Or probably the naming KV is misleading?

Reply:
This is the only Side Input iterating KV API in the Go SDK at present.

While [generics in Go are making progress](https://blog.golang.org/generics-next-step) it's not something we can use at this time. Maybe late next year, we'll be able to add a beam.KV type.

Further, sideinputs are not guaranteed grouped or ordered in any way, so we'd be getting pairs in one form or another regardless.

Reply:
Is the state of Side Input iterating of KV in the Go SDK currently preventing this kata from being helpful?  Should we hold off on a Side Input lesson at this time or change the example to a singleton Side Input like what is done in the documentation?

Reply:
Good question!
No. 
It's a valid way to iterate through KV PCollections as Side inputs. Eg. You know your set of KVs is short, but dynamic, and you'll need all of them at random while processing a DoFn. 

Ideally we also have an alternate Map/Get style random access to KV side inputs as well, but it would be in addition to the existing way.

With generics for Go taking their [next steps](https://blog.golang.org/generics-next-step), maybe next year we could bolster the iteration API with any official Iteration format, in addition to the current methods.

Reply:
@lostluck / @henryken I'll keep close to the corresponding Java kata and use the KV<string, string> form as a source of the side input.

Reply:
Thanks for the explanation @lostluck.
I think we can keep it as it is.